### PR TITLE
move mutability within IC struct to limit borrow panics

### PR DESCRIPTION
**Pull request by Ryex**

### CODE REVIEW

File: ic10emu/src/interpreter.rs [hunk 1]
```diff
@@ -2590,3 +2600,83 @@ pub fn i64_to_f64(i: i64) -> f64 {
...
+            let device = vm.devices.get(&ic).unwrap();
+            let device_ref = device.borrow();
+            device_ref.ic.unwrap()
+        };
+        let ic_chip = vm.ics.get(&ic_id).unwrap().borrow();
+        vm.set_code(
+            ic,
+            r#"lb r0 HASH("ItemActiveVent") On Sum
+            #lb r1 HASH("ItemActiveVent") On Maximum
+            lb r2 HASH("ItemActiveVent") On Minimum"#,
```
Comment (Emilgardis) [inline ic10emu/src/interpreter.rs:2624]: should maybe use a prefab that actually is a prefab, I just picked whatever the autocomplete gave me when I made the issue 😅

Comment (Ryex) [reply ic10emu/src/interpreter.rs:2624]: true, but then again the test is intentionally designed to look for the result where no devices were found

**Comment by Emilgardis** [comment]
Seems like it's not fully working.

```mips
lb r1 0 Maximum Sum
add r0 r0 1
j 0
```

succeeds when stepping, but fails when using run

File: ic10emu/src/vm.rs [hunk 2]
```diff
@@ -378,21 +374,23 @@ impl VM {
             .get(&ic_id)
             .ok_or(VMError::UnknownIcId(ic_id))?
             .clone();
-        ic.borrow_mut().ic = 0;
+        ic.borrow().ic.replace(0);
         self.set_modified(id);
         for _i in 0..128 {
             if let Err(err) = ic.borrow_mut().step(self, ignore_errors) {
```
Comment (Emilgardis) [inline ic10emu/src/vm.rs:380]: Changing this to `borrow` instead of `borrow_mut` resolves the issue I previously mentioned

Comment (Ryex) [reply ic10emu/src/vm.rs:380]: fixed in the other PR

**Pull request merged**